x/oracle/internal/types/expected_keeper.go

```diff
@@ -33,3 +33,8 @@ type SupplyKeeper interface {
 type StakingHooks interface {
 	AfterValidatorBonded(ctx sdk.Context, consAddr sdk.ConsAddress, valAddr sdk.ValAddress) // Must be called when a validator is bonded
 }
+
```

Comment:
Why do we need oracle hooks?

Reply:
To keep oracle swap rate for safe base pool update, but removed. It will retry until success when there is no oracle price.